service/lib/dinstaller/dbus/users.rb

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "dinstaller/users"
+
+module DInstaller
+  module DBus
+    # YaST D-Bus object (/org/opensuse/YaST/Installer1)
+    #
+    # @see https://rubygems.org/gems/ruby-dbus
+    class Users < ::DBus::Object
+      PATH = "/org/opensuse/DInstaller/Users1".freeze
+      private_constant :PATH
+
+      LANGUAGE_INTERFACE = "org.opensuse.DInstaller.Users1".freeze
+      private_constant :USERS_INTERFACE
+
+      def initialize(logger)
+        @logger = logger
+        @root_password_set = false
+
+        super(PATH)
+      end
+
+      dbus_interface USERS_INTERFACE do
+        dbus_reader :root_password_set, "b"
+
+        dbus_reader :root_ssh_key, "s", dbus_name: "RootSSHKey"
+
+        dbus_reader :first_user, "(ssba{sv})"
+
+        dbus_method :SetRootPassword, "in Value:s, in Encrypted:b" do |value, encrypted|
+          logger.info "Setting Root Password"
+          backend.assign_root_password(value, encrypted)
+
+          self[DBus::PROPERTY_INTERFACE].PropertiesChanged(USERS_INTERFACE, {"RootPasswordSet" => !value.empty?}, [])
```

Comment:
Signaling that the password is set is fine; I wonder if `SetRootPassword` should return something.

Reply:
well, only think that comes to my mind is encrypted password, but I am not sure if we have it, as it depends on default encryption method. What else you expect from it?

Reply:
Perhaps if it was successful or not. But I guess that it is something we should discuss for other endpoints too, so let's keep it as it is by now.